.github/actions/setup-nix/action.yml

```diff
@@ -0,0 +1,66 @@
+name: setup-nix
+
+inputs:
+  system:
+    type: string
+    required: true
+  sandbox:
+    type: string
+    default: "true"
+
+runs:
+  using: composite
+  steps:
+    - name: reclaim space (linux)
+      if: runner.os == 'Linux'
+      uses: wimpysworld/nothing-but-nix@main
+      with:
+        hatchet-protocol: rampage
+
+    - name: reclaim space (darwin)
+      if: runner.os == 'macOS'
+      shell: bash
+      run: |
+        echo "::group::disk space (before)"
+        sudo df -h
+        echo "::endgroup::"
+
+        echo "::group::disable mds"
+        sudo mdutil -i off -a || echo "mdutil failed"
+        sudo launchctl unload -w /System/Library/LaunchDaemons/com.apple.metadata.mds.plist \
+         || echo "launchctl unload failed"
+        echo "::endgroup::"
+
+        sudo rm -rf \
+          /Applications/Xcode_* \
+          /Library/Developer/CoreSimulator \
+          /Library/Frameworks \
+          /Users/runner/.dotnet \
+          /Users/runner/.rustup \
+          /Users/runner/Library/Android \
+          /Users/runner/Library/Caches \
+          /Users/runner/Library/Developer/CoreSimulator \
+          /Users/runner/hostedtoolcache &
+
+    - name: install nix
+      uses: cachix/install-nix-action@v31
+      with:
+        # Putting build-dir in /nix is a workaround for https://github.com/wimpysworld/nothing-but-nix/issues/18
+        extra_nix_config: |
+          build-dir = /nix/build
+          sandbox = ${{ inputs.sandbox }}
+          system = ${{ inputs.system }}
+
+    - name: create build-dir
+      shell: bash
+      run: sudo mkdir -p /nix/build
+
+    - name: post setup-nix
+      if: runner.os == 'macOS'
+      uses: srz-zumix/post-run-action@v2
+      with:
+        shell: bash -e {0}
+        post-run: |
+          echo "::group::disk space (after)"
+          sudo df -h
+          echo "::endgroup::"
```

Comment:
Again, if this is in its own step, I _believe_ it gets automatically grouped within the composite step. So explicitly adding a group is probably not needed.

I could be mis-remembering though, and it's worth checking if the explicitly named group offers a better UX than just using the implicit step's group.

Reply:
The post-step shows up as a separate "post setup-nix" step at the end of the workflow.

I think it's fine with the group, but it _is_ the only thing in the post step's logs